memory/level_up_manager.py

```diff
@@ -0,0 +1,123 @@
+from enum import Enum
+from typing import Self
+
+from memory.core import mem_handle
+from memory.mappers.player_party_character import PlayerPartyCharacter
+
+
+class LevelUpUpgradeType(Enum):
+    NONE = -1
+    HitPoint = 0
+    SkillPoint = 1
+    # ComboPoint  = 2
+    PhysicalAttack = 3
+    PhysicalDefense = 4
+    MagicAttack = 5
+    MagicDefense = 6
+
+
+class LevelUpUpgrade:
+    def __init__(self: Self, upgrade_type: LevelUpUpgradeType, active: bool) -> None:
+        self.upgrade_type = upgrade_type
+        self.active = active
+
+
+# This is actually called LevelUpSceneController, but its a manager
+# class, so we'll call it that here.
+class LevelUpManager:
+    NULL_POINTER = 0xFFFFFFFF
+    ZERO_NULL_POINTER = 0x0
+    ITEM_OBJECT_OFFSET = 0x8
+    ITEM_INDEX_0_ADDRESS = 0x20
+
+    def __init__(self: Self) -> None:
```

Comment:
Suggest adding type hints to these, especially the "publicly" exposed values that are assigned `None` or `[]`. So for example:

```python
self.current_upgrades: list[LevelUpUpgrade] = []
```